test/jit_hooks/test_jit_hooks.cpp

```diff
@@ -0,0 +1,270 @@
+#include <torch/script.h>
+
+
+#include <memory>
+#include <string>
+#include <sstream>
+#include <vector>
+
+#include <iostream>
+
+void test_submodule_multiple_hooks_single_input(
```

Comment:
I think there is opportunity for refactoring here as well - a lot of the tests look the same. You could aggregate model paths, inputs, and outputs into some list or something and then run the same test code (load, run, compare output) for all `(file, input, output)` tuples. I think you should be able to deal with varying input/output types by expressing them as `IValues` and relying on `IValue::equals` for comparison.